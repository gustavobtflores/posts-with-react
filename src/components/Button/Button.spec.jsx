import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "./index";

describe("<Button />", () => {
  it("should render the button with text 'Load more posts'", () => {
    render(<Button text={"Load more posts"} />);
    expect.assertions(1);

    const button = screen.getByRole("button", { name: /load more posts/i });
    expect(button).toBeInTheDocument();
  });

  it("should call function on button click", () => {
    const fn = jest.fn();
    render(<Button text={"Load more posts"} onClick={fn} />);

    const button = screen.getByRole("button", { name: /load more posts/i });

    userEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it("should be disabled when flag is true", () => {
    render(<Button text="Load more posts" disabled={true} />);
    const button = screen.getByRole("button", { name: /load more posts/i });

    expect(button).toBeDisabled();
  });

  it("should be enable when flag is false", () => {
    render(<Button text="Load more posts" disabled={false} />);
    const button = screen.getByRole("button", { name: /load more posts/i });

    expect(button).toBeEnabled();
  });
});
