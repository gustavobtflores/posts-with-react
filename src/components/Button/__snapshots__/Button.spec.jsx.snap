// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  class="sc-bczRLJ hZmcDf"
  disabled=""
>
  Load more posts
</button>
`;
